-1 
Tested at: e41f093781c30a567ba1495220a2f7c088243af3
 I found an error when building: 
 
 <pre>>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoLocalTracker/SiStripRecHitConverter/src/StripCPEfromTemplate.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoLocalTracker/SiStripRecHitConverter/src/StripCPEfromTrackAngle.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoLocalTracker/SiStripRecHitConverter/src/StripCPEgeometric.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoLocalTracker/SiStripRecHitConverter/src/VVIObj.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoLocalTracker/SiStripRecHitConverter/src/StripCPE.cc: In member function 'void StripCPE::fillParams()':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoLocalTracker/SiStripRecHitConverter/src/StripCPE.cc:123:77: error: 'geometry' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     p.moduleGeom = SiStripDetId(stripdet->geographicalId()).moduleGeometry();
                                                                             ^
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoTracker/TkDetLayers/src/BladeShapeBuilderFromDet.cc 
cc1plus: some warnings being treated as errors
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_6_X_2015-08-28-1100/src/RecoTracker/TkDetLayers/src/BoundDiskSector.cc 
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10963/7725/summary.html

--- a/DataFormats/SiStripDetId/interface/SiStripDetId.h
+++ b/DataFormats/SiStripDetId/interface/SiStripDetId.h
@@ -163,7 +163,7 @@ SiStripDetId::ModuleGeometry SiStripDetId::moduleGeometry() const {
     case 7: geometry = W7;
       break;
     }
-    case UNKNOWN: default:;  
+  case UNKNOWN: default:;  
   }
   return geometry;
 }

--- a/DataFormats/SiStripDetId/interface/SiStripDetId.h
+++ b/DataFormats/SiStripDetId/interface/SiStripDetId.h
@@ -132,7 +132,7 @@ SiStripDetId::SubDetector SiStripDetId::subDetector() const {
 }
 
 SiStripDetId::ModuleGeometry SiStripDetId::moduleGeometry() const {
-  SiStripDetId::ModuleGeometry geometry;
+  SiStripDetId::ModuleGeometry geometry = UNKNOWNGEOMETRY;
   switch(subDetector()) {
   case TIB: geometry = int((id_>>layerStartBit_) & layerMask_)<3? IB1 : IB2;
     break;
@@ -163,8 +163,7 @@ SiStripDetId::ModuleGeometry SiStripDetId::moduleGeometry() const {
     case 7: geometry = W7;
       break;
     }
-    break;
-  case UNKNOWN: default: geometry = UNKNOWNGEOMETRY;
+    case UNKNOWN: default:;  
   }
   return geometry;
 }

